Corgibytes.Freshli.Cli/CommandRunners/AnalyzeRunner.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.CommandLine.Invocation;
+using System.CommandLine.IO;
+using System.Linq;
+using Corgibytes.Freshli.Cli.CommandOptions;
+using Corgibytes.Freshli.Cli.Commands;
+using Corgibytes.Freshli.Cli.Functionality;
+using Corgibytes.Freshli.Cli.Functionality.Analysis;
+using Corgibytes.Freshli.Cli.Functionality.Engine;
+using Corgibytes.Freshli.Cli.Functionality.FreshliWeb;
+using Corgibytes.Freshli.Lib;
+
+namespace Corgibytes.Freshli.Cli.CommandRunners;
+
+public class AnalyzeRunner : CommandRunner<AnalyzeCommand, AnalyzeCommandOptions>
+{
+    private readonly IApplicationActivityEngine _activityEngine;
+    private readonly ICacheManager _cacheManager;
+    private readonly IApplicationEventEngine _eventEngine;
+    private readonly IResultsApi _resultsApi;
+
+    public AnalyzeRunner(
+        IServiceProvider serviceProvider, Runner runner, IApplicationActivityEngine activityEngine,
+        ICacheManager cacheManager, IApplicationEventEngine eventEngine, IResultsApi resultsApi
+    ) : base(serviceProvider, runner)
+    {
+        _activityEngine = activityEngine;
+        _cacheManager = cacheManager;
+        _eventEngine = eventEngine;
+        _resultsApi = resultsApi;
+    }
+
+    public override int Run(AnalyzeCommandOptions options, InvocationContext context)
+    {
+        _activityEngine.Dispatch(new StartAnalysisActivity(_cacheManager, new HistoryIntervalParser())
+        {
+            CacheDirectory = options.CacheDir,
+            HistoryInterval = options.HistoryInterval,
+            RepositoryBranch = options.Branch,
+            RepositoryUrl = options.RepositoryLocation,
+            UseCommitHistory = options.CommitHistory,
+            GitPath = options.GitPath
+        });
+
+        _eventEngine.On<AnalysisStartedEvent>(startEvent =>
+        {
+            context.Console.Out.WriteLine("Results will be available at: " + _resultsApi.GetResultsUrl(startEvent.AnalysisId));
+        });
+
+        _eventEngine.On<LibYearComputedEvent>(computedEvent =>
+        {
+            var libYearSummed = 0.0;
+            if (computedEvent.LibYearPackages != null)
+            {
+                libYearSummed = computedEvent.LibYearPackages.Sum(libYear => libYear.LibYear);
+            }
+
+            context.Console.Out.WriteLine($"Libyear at {computedEvent.AnalysisLocation?.CommitId} is {libYearSummed}");
```

Comment:
I like that this information is being written out. It'll help show progress as the command is running along. I think the output would be more meaningful if it included the path to the manifest that's being analyzed. Many git repositories are going to have more than one dependency manifest. In those cases, you'd get this output twice for a given CommitId. Also the date maters, too... so that should be included as well. So I think the output would need to be something like

"LibYear for {manifest} at {CommitId} on {date} is {libYearSummed}". What do you think? We can also create an issue for this and have it wait until later. I wouldn't think that this is needed before we demo.

We could also write this line to the log as `Information` if it's going to make the console output too noisy. If we write a detailed message to the log, then I think we could just write out a `.` to the console to show that progress is being made.

Reply:
Haha, yeah I had the same conclusion but I didn't want to add it in this PR. Same with the acceptance tests, I wanted to wait until everything is connected before finalizing them as the tests depend on this output. 

I like the idea of logging it to a log instead of the console output. Maybe add a `--verbose` option to see these sort of logs?

Reply:
Continuing in issue: https://github.com/corgibytes/freshli-cli/issues/240

Reply:
Also related: https://github.com/corgibytes/freshli-cli/issues/241